Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidReader.cs

```diff
@@ -508,8 +510,15 @@ private static void ReadNotes(TrackChunk track, IList<MessageProcessParams> mess
             else
                 unrecognised.UpdateCache();
 
+            // Apply SysEx events first
+            // These are separate to prevent forcing issues on open notes marked via SysEx
+            foreach (var process in processParams.sysexProcessList)
```

Comment:
This could use further clarification, i.e. it mentions forcing issues but doesn't detail what those issues are exactly. If you could add some further comments explaining what happens if they are processed the other way around for example that would be appreciated.

Reply:
Got a whole paragraph put in lol, this bug was a bit of a complex interaction. The short of it is, ensure fret changes happen *before* forcing is applied, or else natural HOPO states can be wrong.

Reply:
Looks good, thanks!